### PR TITLE
FRW-414 Fixed event triggering for plugins returning no data.

### PR DESCRIPTION
**Pull request by fabiospryker**
- Developer(s): @fabiospryker

- Ticket: https://spryker.atlassian.net/browse/FRW-414

- Release Group: https://release.spryker.com/release-groups/view/4459

- merge: merge

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   EventBehavior         | minor                 |                       |

-----------------------------------------

##### Change log

Improvements

- Adjusted `EventResourceRepositoryManager` so that publishing events are triggered for plugins returning no data.

### CODE REVIEW

**Review by dimitriyTsemma — COMMENTED** [review]
please check error in RG regarding this PR
and also check errors in this PR like 
`Parameter #1 $plugin of method Spryker\Zed\EventBehavior\Business\Model\EventResourceRepositoryManager::isPluginWithData() expects Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourceBulkRepositoryPluginInterface, Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourceRepositoryPluginInterface given.
`

**Comment by fabiospryker** [comment]
Spryker CV error seems unrelated: `https://spryker.slack.com/archives/C0521M64S/p1631103007016600`

'1 logical constraint issues: Missing Transfer constraint `^3.25.0` in module `EventBehavior` for new getOrFail functionality.'.

File: src/Spryker/Zed/EventBehavior/Business/Model/EventResourceRepositoryManager.php [hunk 1]
```diff
@@ -197,4 +209,24 @@ protected function triggerBulk(EventResourcePluginInterface $plugin, array $ids)
...
+    {
+        $this->eventFacade->trigger($plugin->getEventName(), new EventEntityTransfer());
+    }
+
+    /**
+     * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourceBulkRepositoryPluginInterface|\Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourceRepositoryPluginInterface $plugin
+     *
+     * @return bool
+     */
+    protected function isPluginWithData($plugin): bool
```
Comment (dimitriyTsemma) [inline src/Spryker/Zed/EventBehavior/Business/Model/EventResourceRepositoryManager.php:228]: this function wrong 
```
EventResourceBulkRepositoryPluginInterface::getData(int $offset, int $limit): array
EventResourceRepositoryPluginInterface::getData(array $ids = []): array
```

**Review by dimitriyTsemma — CHANGES_REQUESTED** [review]

**Review by dimitriyTsemma — CHANGES_REQUESTED** [review]
RG still with errors (because this PR template is wrong)

File: src/Spryker/Zed/EventBehavior/Business/Model/EventResourceRepositoryManager.php (hunk 1 above)
Comment (fabiospryker) [reply src/Spryker/Zed/EventBehavior/Business/Model/EventResourceRepositoryManager.php]: Thanks Dmitry, updated code / logic by comparing it again with commit  7357927e41587d330b520a078761996c2dcd1454 of file EventResourceRepositoryManager.php (git history). In that commit, `triggerPluginOnce` method is used when a plugin is returning no data.

**Comment by fabiospryker** [comment]
> RG still with errors (because this PR template is wrong)

Thanks, sorry was a bit still confused about how to use Release App since this is not usual spryker/spryker repo, updated the template.

**Review by dimitriyTsemma — APPROVED** [review]

**Review by asmarovydlo — APPROVED** [review]

**Review by asmarovydlo — APPROVED** [review]

**Pull request merged**